internal/credentials/kubernetes/ecr/pod_identity.go

```diff
@@ -82,17 +83,22 @@ func (p *podIdentityCredentialHelper) getCredentials(
 	repoURL string,
 	_ *corev1.Secret,
 ) (*credentials.Credentials, error) {
-	if credType != credentials.TypeImage ||
+	if credType == credentials.TypeGit ||
 		p.awsAccountID == "" { // Pod Identity isn't set up for this controller
 		// This helper can't handle this
 		return nil, nil
 	}
 
+	if credType == credentials.TypeHelm && !strings.HasPrefix(repoURL, "oci://") {
+		// Only OCI Helm repos are supported in ECR
+		return nil, nil
+	}
+
 	matches := ecrURLRegex.FindStringSubmatch(repoURL)
-	if len(matches) != 2 { // This doesn't look like an ECR URL
+	if len(matches) != 3 { // This doesn't look like an ECR URL
 		return nil, nil
 	}
-	region := matches[1]
+	region := matches[2]
```

Comment:
```suggestion
	region := matches[1]
```

Complements https://github.com/akuity/kargo/pull/2280/files#r1674295123